### PR TITLE
staging-next fixups

### PR DESCRIPTION
**Pull request by lukegb**
###### Motivation for this change

Try to get staging-next closer to being ready to merge into master.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/aws-sam-cli/default.nix [hunk 1]
```diff
@@ -44,7 +44,7 @@ python3.pkgs.buildPythonApplication rec {
       --replace "Flask~=1.1.2" "Flask~=2.0" \
       --replace "dateparser~=0.7" "dateparser>=0.7" \
       --replace "docker~=4.2.0" "docker>=4.2.0" \
-      --replace "requests==" "requests~=" \
+      --replace "requests==" "requests #" \
```
Comment (SuperSandro2000) [inline pkgs/development/tools/aws-sam-cli/default.nix:47]: Why not use >=

**Comment by vcunat** [comment]
Oh, this rebuilds darwin stdenv...

**Comment by lukegb** [comment]
Not only that, but it broke aarch64-darwin _sigh_

**Comment by vcunat** [comment]
Ah, I didn't notice that the last `staging-next` commit wasn't part of this PR, so that regression is hopefully fixed now.

File: pkgs/development/tools/aws-sam-cli/default.nix (hunk 1 above)
Comment (lukegb) [reply pkgs/development/tools/aws-sam-cli/default.nix:47]: This is more similar to the one below it (`==` -> ` #`, than `~=` -> `>=` ). No particular reason other than that.

Comment (SuperSandro2000) [reply pkgs/development/tools/aws-sam-cli/default.nix:47]: This also removes potential conditions after the requirement. I would just stick to >=.

**Pull request merged**